### PR TITLE
Improve Phlex test coverage

### PR DESCRIPTION
**Pull request by greenc-FNAL**
Factored out of #329.

- **Improve test coverage of Phlex code**

### CODE REVIEW

File: test/identifier.cpp [hunk 1]
```diff
@@ -65,5 +65,15 @@ int main()
     }
   }
   assert(ok);
+
+  // Additional coverage for identifier edge cases
+  identifier id1("abc");
+  identifier id2("def");
+  assert(id1 == id1);
+  assert(id1 != id2);
+  assert(id1 < id2 || id2 < id1);
```
[the remark this answers is not in the document]
Comment (beojan) [reply test/identifier.cpp]: Not if the operators are defined incorrectly.
